com.unity.render-pipelines.high-definition/Runtime/Sky/VisualEnvironment.cs

```diff
@@ -74,4 +81,146 @@ public sealed class SkyAmbientModeParameter : VolumeParameter<SkyAmbientMode>
+            {
+                if (mode == WindOverrideMode.Global)
+                    return mode.ToString();
+                string str = null;
+                if (mode == WindOverrideMode.Custom)
+                    str = customValue.ToString();
+                if (mode == WindOverrideMode.Additive)
+                    str = additiveValue.ToString();
+                if (mode == WindOverrideMode.Multiply)
+                    str = multiplyValue.ToString();
+                return str + " (" + mode.ToString() + ")";
+            }
+        }
+
+        /// <summary>Wind volume parameter constructor.</summary>
+        /// <param name="value">Initial value.</param>
+        /// <param name="value">Initial override mode.</param>
+        /// <param name="overrideState">Initial override state.</param>
+        public WindParameter(float value = 0.0f, WindOverrideMode mode = WindOverrideMode.Global, bool overrideState = false)
+            : base(default, overrideState)
+        {
+            this.value = new WindParamaterValue
+            {
+                mode = mode,
+                customValue = mode <= WindOverrideMode.Global ? value : 0.0f,
+                additiveValue = mode == WindOverrideMode.Additive ? value : 0.0f,
+                multiplyValue = mode == WindOverrideMode.Multiply ? value : 1.0f,
+            };
+        }
+
+        /// <summary>Interpolates between two values.</summary>
+        /// <param name="from">The start value</param>
+        /// <param name="to">The end value</param>
+        /// <param name="t">The interpolation factor in range [0,1]</param>
+        public override void Interp(WindParamaterValue from, WindParamaterValue to, float t)
+        {
+            m_Value.mode = t > 0f ? to.mode : from.mode;
+            m_Value.customValue = from.customValue + (to.customValue - from.customValue) * t;
+            m_Value.additiveValue = from.additiveValue + (to.additiveValue - from.additiveValue) * t;
+            m_Value.multiplyValue = from.multiplyValue + (to.multiplyValue - from.multiplyValue) * t;
+        }
+
+        /// <summary>Returns a hash code for the current object.</summary>
+        /// <returns>A hash code for the current object.</returns>
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                int hash = 17;
+                hash = hash * 23 + overrideState.GetHashCode();
+                hash = hash * 23 + value.mode.GetHashCode();
+                hash = hash * 23 + value.customValue.GetHashCode();
+                hash = hash * 23 + value.additiveValue.GetHashCode();
+                hash = hash * 23 + value.multiplyValue.GetHashCode();
+
+                return hash;
+            }
+        }
+
+        /// <summary>Returns interpolated value from the visual environment.</summary>
+        /// <returns>The value for this parameter.</returns>
+        public float GetValue(HDCamera camera)
```

Comment:
Not a big fan of doing volume evaluation inside this nested function, we should pass it the global windValue directly as a parameter instead of passing the camera and evaluating the volume component.

Reply:
do we really want to evaluate the volume ourselves when getting the value ? plus that means we would evaluate it for nothing when in custom mode (not a big deal but still)